testsuite/tests/basic/patmatch.ml

```diff
@@ -149,8 +149,8 @@ let () =
 
 (* No string sharing PR#6322 *)
 let test x = match x with
-  | true -> "a"
-  | false -> "a"
+  | true -> Bytes.of_string "a"
+  | false -> Bytes.of_string "a"
```

Comment:
@maranget, @chambart - does adding `Bytes.of_string` here completely defeat the purpose of this test?